File: LRSMessageToolBar/Classes/Helper/LRSMessageToolBarHelper.m
//
//  LRSMessageToolBarHelper.m
//  LRSMessageToolBar-LRSMessageToolBar
//
//  Created by sama 刘 on 2021/10/27.
//

#import "LRSMessageToolBarHelper.h"

@implementation LRSMessageToolBarHelper

static CGFloat kLRSMessageToolBarHelperScale = 1;
static bool kLRSIPhoneXSeriesDevice = true;

+ (CGFloat)screenWidth {
    return [UIScreen mainScreen].bounds.size.width;
}

+ (CGFloat)screenHeight {
    return [UIScreen mainScreen].bounds.size.height;
}

+ (CGFloat)scale {
    return kLRSMessageToolBarHelperScale;
}

+ (void)setScale:(CGFloat)scale {
    kLRSMessageToolBarHelperScale = scale;
}

+ (NSBundle *)bundle {
    static NSBundle *sourceBundle = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        NSBundle *bundle = [NSBundle bundleForClass:[LRSMessageToolBarHelper class]];
        NSURL *url = [bundle URLForResource:@"LRSMessageToolBar" withExtension:@"bundle"];
        sourceBundle = [NSBundle bundleWithURL:url];
    });
    return sourceBundle;
}

+ (UIImage *)imageNamed:(NSString *)named {
    return [UIImage imageNamed:named inBundle:[self bundle] compatibleWithTraitCollection:nil];
}

+ (UIColor *)colorNamed:(NSString *)named {
    return [UIColor colorNamed:named inBundle:[self bundle] compatibleWithTraitCollection:nil];
}

+ (NSArray<LRSMemePackageConfigure *> *)allEmojis {
    NSString *plistName = nil;
    if (@available(iOS 12.1, *)) {
        plistName = @"EmojisList_ios>12.1";
    } else if (@available(iOS 11.1, *)) {
        plistName = @"EmojisList_ios>=11";
    } else {
        plistName = @"EmojisList_ios<11";
    }
    NSString *plistPath = [[self bundle] pathForResource:plistName ofType:@"plist"];
    NSArray *arr = [NSArray arrayWithContentsOfFile:plistPath];
    NSMutableArray *emojis = [NSMutableArray arrayWithCapacity:arr.count];
    for (int i = 0; i < arr.count; i++) {
        LRSMemePackageConfigureItem *emoji = [[LRSMemePackageConfigureItem alloc] init];
        emoji.emojiValue = [arr objectAtIndex:i];
        [emojis addObject:emoji];
    }
    LRSMemePackageConfigure *configure = [[LRSMemePackageConfigure alloc] init];
    configure.emojis = emojis;
    configure.key = @"default";
    configure.coverImageName = @"information_expression_normal";
    configure.row = 4;
    configure.column = 9;
    return @[configure];
}

+ (CGSize)sizeForEmojiKey:(NSString *)key {
    static CGSize size;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        size = CGSizeMake(45, 37);
    });
    return size;
}

+ (void)setIPhoneXSeriesDevice:(BOOL)iPhoneXSeriesDevice {
    kLRSIPhoneXSeriesDevice = iPhoneXSeriesDevice;
}

+ (BOOL)isIPhoneXSeriesDevice {
    return kLRSIPhoneXSeriesDevice;
}
@end

File: LRSMessageToolBar/Classes/Helper/LRSMessageToolBarHelper.swift
//
//  LRSMessageToolBarHelper.swift
//  LRSMessageToolBarComponents
//
//  Created by sama 刘 on 2021/11/16.
//

import UIKit
import Then

class LRSMessageToolBarHelper: NSObject {
    static var scale: Float = 1
    static var safeAreaHeight: CGFloat = 0

    static func emojis() -> [LRSMemePackageConfigure]? {
        var plistName: String
        if #available(iOS 12.1, *) {
            plistName = "EmojisList_ios>12.1"
        } else if #available(iOS 11.1, *) {
            plistName = "EmojisList_ios>=11"
        } else {
            plistName = "EmojisList_ios<11";
        }
        guard let path = Bundle.now?.path(forResource: plistName, ofType: "plist") else {
            return nil
        }
        guard let values = NSArray(contentsOfFile: path) as? [String] else {
            return nil
        }
        let emojis = values.map { value in
            return LRSMemePackageConfigure.Item(emojiValue: value)
        }
        let configure = LRSMemePackageConfigure(key: "default", emojis: emojis, coverImageName: "information_expression_normal", row: 4, column: 9)
        return [configure]
    }

    static func sizeOfEmoji(key: String) -> CGSize {
        return CGSize(width: 45, height: 37)
    }

}


internal extension UIScreen {
    var width: CGFloat {
        return self.bounds.size.width
    }

    var height: CGFloat {
        return self.bounds.size.height
    }
}

internal extension Bundle {
    static var now: Bundle? {
        let bundle = Bundle(for: LRSMessageToolBarHelper.self)
        guard let url = bundle.url(forResource: "LRSMessageToolBar", withExtension: "bundle") else {
            return .main
        }
        return Bundle(url: url)
    }
}


internal extension UIImage {
    static func image(named: String) -> UIImage? {
        if #available(iOS 13.0, *) {
            return UIImage(named: named, in: .now, with: nil)
        } else {
            return UIImage(named: named, in: .now, compatibleWith: nil)
        }
    }
}


internal extension UIColor {
    static func color(named: String) -> UIColor? {
        return UIColor(named: named, in: .now, compatibleWith: nil)
    }
}
